textBasedAdventure.js

```diff
@@ -1,5 +1,325 @@
+// const readLineSync = require("readline-sync");
+
+// const name = readLineSync.question("What's your name? ");
+// console.log("Hello " + name + " it's nice to meet you!")
+// Whoever is grading my story, that first problem was funny
+
+let nameInput = readline1.question("Enter your name: ");
+
+let doorChoice;
+
+const intro = () => {
+  console.log("Welcome to my winter adventure, " + nameInput + "!");
+  console.log(
+    "To play, just read the text and answer the questions when you're asked to make a choice."
+  );
+  console.log("");
+  console.log("You are in a room with three doors: a, b and c");
+};
+
+intro();
+
+const doorLocked = () => {
+  doorChoice = readline1.question("Which door do you choose (a, b or c)? ");
+  if (doorChoice === "a") {
+    console.log("\nThis door is locked. Please, try another (b or c).");
+    tryAgain();
+  } else if (doorChoice === "b") {
+    console.log(
+      "\nYou enter a room with a table. \nThere is a cup of hot chocolate on it.");
+    chocQuestion();
+  } else if (doorChoice === "c") {
+    doorChoiceC();
+  } else {
+    console.log("\nPlease make a valid selection (a, b or c).\n");
+    tryAgain();
+  }
+};
+
+const doorChoiceC = () => {
+  console.log(
+    `\nThis door takes you outside. \nYou see a friendly snowperson. \nThe snowperson says: "Hello! What is your name?"`
+  );
+  console.log(
+    "You look at it up and down but don't trust it. You give a fake name."
+  );
+  nameInputB = readline1.question("Tell the snowperson your name:\n");
+  if (nameInputB.toLowerCase() === nameInput.toLowerCase()) {
+    console.log("You are really trusting.");
+  }
+  console.log(`\nThe snowperson says, "Wow, ${nameInputB} is my name too!"`);
+  console.log(
+    `"Whoa, I just got déjà vu. \nAnyway have this diamond. You are now rich!" \nTHE END\n`
+  );
+  afterMike();
+};
+
+const tryAgain = () => {
+  doorLocked();
+};
+
+const chocQuestion = () => {
+  if (readline1.keyInYNStrict("Do you want to drink the hot cocoa? ")) {
+    console.log(
+      "\nYou feel very warm and content. Now you are ready to play outside in the snow. You leave this room and go back where you came."
+    );
+    tryAgain();
+  } else {
+    console.log(
+      "\nThere is nothing else to do here and you start to feel very cold. \nWouldn't a nice warm cup of hot chocolate warm you up?"
+    );
+    chocQuestion();
+  }
+};
+const afterMike = () => {
+  console.log("Or is it...\n");
+  console.log("\nIt's not. :/\n");
+  tapRub();
+};
+
+const tapRub = () => {
+  console.log("Tap your head and rub your belly simultaneously");
+  canYouTapRub();
+};
+
+const canYouTapRub = () => {
+  if (readline1.keyInYNStrict("Can you complete this challnge? ")) {
+    console.log(
+      "\nExcellent work! \nThe troll thats hiding under a bridge that you didn't notice allows you passage."
+    );
+    console.log(
+      `You say to yourself, "Self, if there's a bridge, I'm gonna cross it!"`
+    );
+    spirit = readline1.question(
```

Comment:
remember to always declare your variables. Otherwise you're making a global variable which you probably don't want. Break the habit by adding "use strict" at the top of your files.